README.md

```diff
@@ -82,20 +82,27 @@ More advanced changes can take effect using the `run.d` system. Similar to the `
 
 #### Sequence of events for a crashed supervised service: 
 
-1. [finish](https://github.com/just-containers/s6-overlay#writing-an-optional-finish-script) script is executed
+1. S6 [finish](https://github.com/just-containers/s6-overlay#writing-an-optional-finish-script) scripts are executed. The supervised service is presumed down at this point, its return code is a variable (`${1}`).
 1. If no `finish` script is specified, service gets restarted, with no further action
 1. If `finish` script specifies to bring the container down, admin-initiated container termination behavior applies (below).
 
 
 #### Sequence of events for a `docker stop` or admin-initiated container termination: 
 
-1. SIGTERM is broadcast to all supervised services, described as a [run](https://github.com/just-containers/s6-overlay#writing-a-service-script) script.
-1. Scripts in `/etc/cont-finish.d` are executed, each with S6_KILL_FINISH_MAXTIME
-1. S6 finish scripts are executed, each with S6_KILL_GRACETIME, described as a [finish](https://github.com/just-containers/s6-overlay#writing-an-optional-finish-script) script
-1. SIGHUP is broadcast to all services, in all trees
-1. SIGTERM is broadcast to all services, in all trees
+1. SIGTERM is broadcast to all supervised services, described as a [run](https://github.com/just-containers/s6-overlay#writing-a-service-script) script. `S6_KILL_GRACETIME` corresponds with how long to wait for a SIGTERM to complete and return. This first signal can be trapped/intercepted to convert into a graceful stop (see below). Failure for the supervised service to respond to the signal within `S6_KILL_GRACETIME` will result in an untrappable SIGKILL.
+1. Scripts in `/etc/cont-finish.d` are executed, each with `S6_KILL_FINISH_MAXTIME`.
+1. S6 [finish](https://github.com/just-containers/s6-overlay#writing-an-optional-finish-script) scripts are executed. The supervised service is presumed down at this point, its return code is a variable (`${1}`).
```

Comment:
What is the significance of the variable? How can it be used?

Reply:
We should mention that S6_KILL_GRACETIME applies here. Also, if a finish script takes longer than S6_KILL_GRACETIME, what happens to it? SIGKILL? SIGHUP, SIGTERM, SIGKILL?